### PR TITLE
Plans: Implement plan and product upsells for Jetpack backup product

### PR DESCRIPTION
**Pull request by tyxla**
This PR implements upsell buttons for the Jetpack backup single products. This functionality supports a couple of use cases, and to function it requires a map of product features and their higher tier upsells:

* Suggest a higher-tier product when we have already purchased a lower-tier product.
* Suggest a higher-tier plan with the higher tier product when we have already purchased a plan that includes the lower tier of the product.

#### Changes proposed in this Pull Request

* Plans: Implement plan and product upsells for Jetpack backup product

#### Preview

Higher-tier product upsell:

Upsell with a plan that offers a higher-tier product:

#### Testing instructions

* Checkout this branch.
* For a Jetpack site, verify you have a Jetpack backup daily yearly product.
* Go to http://calypso.localhost:3000/plans/:site where `:site` is the Jetpack site slug.
* Verify you get the "realtime yearly" suggested as an upsell purchase.
* For another Jetpack site, verify you have a Jetpack personal plan.
* Go to http://calypso.localhost:3000/plans/:site where `:site` is the Jetpack site slug.
* Verify you get the Jetpack professional plan suggested as an upsell purchase.
* Verify the devdocs example http://calypso.localhost:3000/devdocs/blocks/product-selector still works well.

### CODE REVIEW

**Comment by enejb** [comment]
I tested this PR and this if what I found. 

1. When you are upgrading to Jepack real time backup and you have the daily backup already. 
The price that you are asked to pay doesn't reflect that you already have paided for the daily backup product. 

2. When you are updating from the personal Plan. And want the real time backups you are not able to just buy the real time backup. Since you already have a personal plan. We recommend to the user
that they buy the Professional plan. Feels weird. Mostly since the user would be able to first buy the real time backups and then buy the personal plan. 

3. If the user has the Professional Plan. Should we hide the option to but any backup product. Right now the option is available to buy any backup product. 

4. We can currentyly switch to the yearly plan and then procceed to buy a product. Since there is no relation ship between the different monthly products.

**Comment by enejb** [comment]
I think everything expect for number 3 should be done in different PRs.

**Comment by tyxla** [comment]
Thanks for thoroughly testing, @enejb, and for the thorough feedback 🙇 

1) This should be addressed in the WP.com store, cc @Automattic/chronos 

2) This is by design, and while I'd like to audit all these cases, I guess this will make most sense for a subsequent iteration.

3) This has just been addressed with the latest commits.

4) I'm planning to work on that in another PR, as it affects not just the cases that this PR introduces, but the entire `ProductSelector` block.

**Comment by tyxla** [comment]
This could use another review now 👁

File: client/blocks/product-selector/README.md [hunk 1]
```diff
@@ -85,6 +85,7 @@ The following props can be passed to the Product Selector block:
 		}
 		```
 	* `optionsLabel`: ( string ) Title of the product options section.
+	* `productUpsells`: ( object ) Optional product upsells. Used to create a map of suggested higher-tier products as an up-sell for a purchased product.
```
Comment (delawski) [inline client/blocks/product-selector/README.md]: We could add here an example of the `productUpsells` structure, similar to what we have for e.g. the `productPriceMatrix`.

**Comment by enejb** [comment]
I did some more testing here. 
And this is what I found. 

If I have the yearly personal plan 
What I now see is the following 
Screen Shot 2019-11-04 at 11 29 40 AM

It is asking me to update to the Professional Plan. But the early one instead of the monthly. 

One thing that I did notice during testing is that we do prompt the user to upgrade to to the yearly plan from monthly. But we never allow the user to switch from monthly to yearly plan. 

I think the issue that I mentioned above can be addressed in a different PR. Since currently I am not sure we will be up selling users on the new plan vs just the product.

File: client/blocks/product-selector/README.md (hunk 1 above)
Comment (tyxla) [reply client/blocks/product-selector/README.md]: Good idea, updated in https://github.com/Automattic/wp-calypso/pull/37250/commits/167ca5fb165d2223f09f880e14eca9a3ae803dfe

**Comment by tyxla** [comment]
> It is asking me to update to the Professional Plan. But the early one instead of the monthly.

Thanks @enejb this is a great catch. I've addressed this in this PR, because right now we still differentiate between the states of different billing intervals and different product options.

As I mentioned above, I'm planning to work on consolidating those in another PR, so it only makes sense to keep the behavior consistent as we iterate.

**Comment by tyxla** [comment]
I've addressed all the feedback, this could use another ✅ from both reviewing folks 😉

**Comment by enejb** [comment]
Nice work @tyxla 
This is definitely helps with all the different Upsell prompts.

**Pull request merged**